sdk/servicebus/azure-messaging-servicebus/src/main/java/com/azure/messaging/servicebus/implementation/ServiceBusManagementSerializer.java

```diff
@@ -35,21 +39,39 @@
     public String serialize(Object object, SerializerEncoding encoding) throws IOException {
         final String contents = jacksonAdapter.serialize(object, encoding);
 
+        final Class<?> clazz = object.getClass();
+        if (!CreateQueueBody.class.equals(clazz) && !CreateRuleBody.class.equals(clazz)
+            && !CreateSubscriptionBody.class.equals(clazz)) {
+            return contents;
+        }
+
         // This hack exists because the service requires a global namespace for the XML rather than allowing
         // each XML element to be prefaced with an explicit namespace. For example:
         // xmlns="foo" works because "foo" is assigned the global namespace.
         // xmlns:ns0="foo", and then prefixing all elements with ns0:AuthorizationRule will break.
-        if (object instanceof CreateQueueBody) {
-            final Matcher matcher = NAMESPACE_PATTERN.matcher(contents);
-            if (matcher.find()) {
-                final String namespace = matcher.group("namespace");
-                return contents
-                    .replaceAll(namespace + ":", "")
-                    .replace("xmlns:" + namespace + "=", "xmlns=");
-            }
+        final Matcher namespaceMatcher = NAMESPACE_PATTERN.matcher(contents);
+        if (!namespaceMatcher.find()) {
+            logger.info("Could not find {} in {}", NAMESPACE_PATTERN.pattern(), contents);
+            return contents;
+        }
+
+        final String namespace = namespaceMatcher.group("namespace");
+        final String replaced = contents
+            .replaceAll(namespace + ":", "")
+            .replace("xmlns:" + namespace + "=", "xmlns=");
+
+        if (!CreateRuleBody.class.equals(clazz)) {
+            return replaced;
         }
 
-        return contents;
+        // This hack is here because RuleFilter and RuleAction type="Foo" should have a namespace like n0:type="Foo".
+        final Matcher filterType = FILTER_ACTION_PATTERN.matcher(replaced);
+        if (filterType.find()) {
+            return filterType.replaceAll("<$1 xmlns:ns0=\"http://www.w3.org/2001/XMLSchema-instance\" ns0:type=");
```

Comment:
Rules might also have ns1 in some cases.

Reply:
Do you know when? I couldn't find an example.

Reply:
This is a Python generated XML that has both ns0 and ns1. Disregard this comment if Java generates differently without ns1.
In the following file, "ns0" doesn't hurt. "ns1" hurts.

```xml
<ns0:entry xmlns:ns0="http://www.w3.org/2005/Atom" xmlns:ns1="http://schemas.microsoft.com/netservices/2010/10/servicebus/connect" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance">
	<ns0:content type="application/xml">
		<ns1:RuleDescription>
			<ns1:Filter xsi:type="SqlFilter">
				<ns1:CompatibilityLevel>20</ns1:CompatibilityLevel>
				<ns1:RequiresPreprocessing>true</ns1:RequiresPreprocessing>
			</ns1:Filter>
			<ns1:Action xsi:type="EmptyRuleAction"/>
			<ns1:Name>rule1</ns1:Name>
		</ns1:RuleDescription>
	</ns0:content>
</ns0:entry>

```

Reply:
Normally validity of XML can be verified at generation time, I am not sure if generator are doing  that.

Reply:
The XML generated by Jackson is valid. It's the service side that is fickle about what XML it likes to consume, that's why we're doing all these hacks.

I verified that we aren't generating multiple namespaces. The reason for this is because we use a global xml namespace, so we don't need n0.